### PR TITLE
Wire up font service on Fuchsia

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

**Comment by abarth** [comment]
(Depends on https://fuchsia-review.googlesource.com/c/9621/)

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp [hunk 1]
```diff
@@ -2,12 +2,130 @@
...
+      return mojo::FontSlant::UPRIGHT;
+    case FontStyleItalic:
+      return mojo::FontSlant::ITALIC;
+  }
+  ASSERT_NOT_REACHED();
+  return mojo::FontSlant::UPRIGHT;
+}
+
+void UnmapMemory(const void* buffer, void* context) {
+  mx_process_unmap_vm(0, reinterpret_cast<uintptr_t>(buffer), 0);
```
Comment (chinmaygarde) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: Are we sure we don't need to pass in the length in the unmap call? I have a suspicion this is a leak. I expect it to unmap a zero address range. But I could be wrong. We probably need a struct that hold the buffer and its length and have that passed as a baton in `MakeWithProc` below.

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp [hunk 2]
```diff
@@ -2,12 +2,130 @@
...
+}
+
+sk_sp<SkData> MakeSkDataFromVMO(mx_handle_t vmo) {
+  uint64_t size = 0;
+  mx_status_t status = mx_vmo_get_size(vmo, &size);
+  if (status < 0 || size > std::numeric_limits<mx_size_t>::max())
```
Comment (chinmaygarde) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: Nit: I feel like `status != MX_NO_ERROR` reads cleaner.

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp [hunk 3]
```diff
@@ -2,12 +2,130 @@
...
+  request->weight = ToMojoWeight(fontDescription.weight());
+  request->width = static_cast<uint32_t>(fontDescription.stretch());
+  request->slant = ToMojoSlant(fontDescription.style());
+
+  mojo::FontResponsePtr response;
+  auto& font_provider = GetFontProvider();
+  font_provider->GetFont(
+      std::move(request),
+      [&response](mojo::FontResponsePtr r) { response = std::move(r); });
+  font_provider.WaitForIncomingResponse();
```
Comment (chinmaygarde) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: An interesting sequence of operations. We should probably check that we got the response for the correct font request. Here we just wait for the first one. Not sure if typefaces can be created on multiple threads.

**Comment by chinmaygarde** [comment]
Concern about a potentially leaky mapping. Other minor nits. But mostly lgtm.

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp (hunk 1 above)
Comment (abarth) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: I asked Travis, and he says that passing zero will remove the whole mapping.  I added that to the docs for this call: https://fuchsia.googlesource.com/magenta/+/master/docs/syscalls/process_unmap_vm.md

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp (hunk 3 above)
Comment (abarth) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: They're all created on the IU thread.  The Mojo bindings will helpfully assert if you try to use an InterfacePtr from multiple threads.

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp (hunk 2 above)
Comment (abarth) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: Done.

File: sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp (hunk 1 above)
Comment (chinmaygarde) [inline sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp]: Perfect. I was afraid the behavior was like `munmap` where you would get an `EINVAL`. Thanks for clearing it up and the docs.

**Comment by chinmaygarde** [comment]
lgtm. Thanks for the clarifications.

**Pull request merged**